### DISCUSSION TITLE
How does lifelines model for left truncation?

### DISCUSSION THREAD

**Discussion by johol355** [CamDavidsonPilon/lifelines#1359] [Q&A]
Hello friends, 

I do understand the bias that left truncation causes. After reading the [documentation](https://lifelines.readthedocs.io/en/latest/Survival%20analysis%20with%20lifelines.html#left-truncated-late-entry-data) several times, I'm still having difficulties understanding _how_ lifelines models for left truncation. Also, I did take a look at the source code without being able to tell what is going on. I would really appreciate any explanation or pointer to understand this matter.

**Comment by CamDavidsonPilon** [comment]
Hi @johol355, are you interested in a particular survival model in particular, like the Kaplan Meier?

**Reply by johol355** [reply]
Hi @CamDavidsonPilon, thanks for taking your time to answer my question. At this point, I've been looking mainly at Kaplan Meier. But the question is relevant for other survival models (e.g. proportional-hazards models) as well. I've just read a paper (doi.org/10.1093/aje/kwq481) on adjusting for left truncation. In the paper, a parametric (normal) distribution of the entry time is posited and used for adjustment. I'm guessing lifelines does something similar?

**Reply by benslack19** [reply]
@johol355 and @CamDavidsonPilon was this ever addressed? I'm currently wondering the same thing for proportional hazards models. We tried looking through the source code and it seems to point to the `CoxTimeVaryingFitter` (like perhaps [here?](https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/fitters/coxph_fitter.py#L1410-L1417)) Does delayed entry in `CoxPHFitter` simply use the logic in the `CoxTimeVaryingFitter`?

**Reply by CamDavidsonPilon** [reply]
Yea, you're correct. The likelihood calculation for late entries requires some extra logic, and the CoxTimeVarying model has the machinery for that, so we just defer to that for the gradient calculations.

**Reply by benslack19** [reply]
Thank you for the confirmation @CamDavidsonPilon !

**Reply by benslack19** [reply]
@CamDavidsonPilon should we expect that a CPH model and a CTV model give exactly the same summary values, all else being equal (one value per subject, etc.)? I'm seeing coefficients match up exactly but not the SEs. Not sure why SEs would be different, especially if left truncation is essentially using the CTV model under the hood.

**Reply by CamDavidsonPilon** [reply]
We have tests that confirm the coefficients should be the same. I'm not sure why the SEs are different though...
https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/tests/test_estimation.py#L5252-L5284

**Reply by CamDavidsonPilon** [reply]
does one model have strictly greater SE than the other?

**Reply by benslack19** [reply]
No, sometimes it's the CPH model and sometimes it's CTV that has a higher SE.  Thanks for sharing the links to the tests.

**Reply by benslack19** [reply]
I believe I figured out what happened after thinking that left truncation was a factor in the difference in standard errors. It wasn't. It was because I was using `cluster_col` in the `CoxPHFitter` and saw in the documentation that the sandwich estimator gets used and that's why the SE changes. I found a few issues while investigating this, though, and I added my observations to issue #544 and created issue #1598 and issue #1599.
